[LT] Add ir_util for ComputePostOrder

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #67282

Test Plan:
`build/bin/test_lazy`

Differential Revision: [D31961754](https://our.internmc.facebook.com/intern/diff/D31961754)

desertfire committed:
[LT] Add ir_util for ComputePostOrder

Test Plan:
`build/bin/test_lazy`

[ghstack-poisoned]

wconstab reviewed (approved):
LGTM, thanks for cleaning this up and landing it!

desertfire committed:
Update on "[LT] Add ir_util for ComputePostOrder"


Test Plan:
`build/bin/test_lazy`

Differential Revision: [D31961754](https://our.internmc.facebook.com/intern/diff/D31961754)

[ghstack-poisoned]

desertfire committed:
Update on "[LT] Add ir_util for ComputePostOrder"


Test Plan:
`build/bin/test_lazy`

Differential Revision: [D31961754](https://our.internmc.facebook.com/intern/diff/D31961754)

[ghstack-poisoned]

desertfire committed:
Update on "[LT] Add ir_util for ComputePostOrder"


Test Plan:
`build/bin/test_lazy`

Differential Revision: [D31961754](https://our.internmc.facebook.com/intern/diff/D31961754)

[ghstack-poisoned]
